Sources/StreamChat/Models/BlockedUser.swift

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A type representing a blocked user. `BlockedUser` is an immutable snapshot of a blocked user entity at the given time.
+///
+public class BlockedUser {
+    /// The unique identifier of the blocked user.
+    public let blockedUserId: UserId
+
+    /// The date the user was blocked.
+    public let blockedAt: Date?
+
+    init(
+        blockedUserId: UserId,
+        blockedAt: Date?
+    ) {
+        self.blockedUserId = blockedUserId
+        self.blockedAt = blockedAt
+    }
+}
+
+extension BlockedUser: Hashable {
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(blockedUserId)
+    }
+}
+
+extension BlockedUser: Equatable {
+    public static func == (lhs: BlockedUser, rhs: BlockedUser) -> Bool {
+        lhs.blockedUserId == rhs.blockedUserId
+            && lhs.blockedAt == rhs.blockedAt
+    }
+}
```

Comment:
Why do we need to it be Equatable for now? 🤔

Reply:
We don't, removed 👍